### ISSUE TITLE
Make puppeteer peerDependency

### ISSUE THREAD

**Issue by stereobooster** [peterbe/minimalcss#108]
If I use `minimalcss` in `react-snap` it will download two copies of `puppeteer`. If we make `minimalcss-core` without CLI and with peerDependency puppeteer and `minimalcss` with CLI and with `puppeteer+minimalcss-core` dependency I will be able to use `minimalcss-core` in `react-snap` with one copy of `puppeteer`

**Comment by peterbe** [comment]
I have only read about workspaces but if it means we can have one `package.json` for the cli and one `package.json` for the "core", that would be awesome. 
But what does it mean in terms of npmjs.com? 
I do like the idea of splitting the code into smaller parts but I'm quite pessimistic about breaking up the repo.

**Comment by stereobooster** [comment]
> I have only read about workspaces but if it means we can have one package.json for the cli and one package.json for the "core", that would be awesome.

Yes.

> But what does it mean in terms of npmjs.com?

It means you will have two separate npm packages, but they will live in one repo for development convenience. 

> I do like the idea of splitting the code into smaller parts but I'm quite pessimistic about breaking up the repo.

[Create-react-app does this](https://github.com/facebook/create-react-app/tree/next/packages), for example.

**Comment by peterbe** [comment]
Neat! I like this. 
Do I need to run `jest` inside each directory? (not a problem)

**Comment by stereobooster** [comment]
> Do I need to run jest inside each directory?

Yes, but there will be one top level `package.json` you can create one task which will run all tests in subdirectories
